ilastik/workflows/carving/watershed_segmentor.py

```diff
@@ -136,3 +138,4 @@ def saveH5G(self, h5g):
 
     def setResulFgObj(self, fgNodes):
         self.gridSegmentor.setResulFgObj(fgNodes)
+        self.hasSeg = True
```

Comment:
It seems like `self.hasSeg` was not properly utilized before (was not in sync sometimes). For some reason I would feel safer if self.hasSeg could be a dynamic property, rather than relying on every  method setting it correctly. But I take it that you ironed it out appropriately.

Reply:
Agree. I'm still not 100% confident this works as intended and refrained from reworking it for now to not bloat this PR even more, but the most obvious issue should be fixed for now...